group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -739,6 +748,19 @@ public Optional<OffsetExpirationCondition> offsetExpirationCondition() {
         return Optional.of(new OffsetExpirationConditionImpl(offsetAndMetadata -> offsetAndMetadata.commitTimestampMs));
     }
 
+    /**
+     * @return The current state as a capitalized String with given committedOffset.
+     */
+    public String stateAsCapitalizedString(long committedOffset) {
+        return state.get(committedOffset).toCapitalizedString();
+    }
+
+    @Override
+    public boolean isInStates(Set<String> statesFilter, long committedOffset) {
+        return statesFilter.contains(this.stateAsCapitalizedString(committedOffset)) || statesFilter.contains(
+            this.stateAsString(committedOffset));
```

Comment:
Same question about checking the two versions here.

Reply:
Done.